packages/plugins/model-train/bayesian-model-train/src/script.ts

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+import fs from 'fs';
+import readline from 'readline';
+import { zip, unzip } from 'lodash';
+
+const boa = require('@pipcook/boa');
+
+const jieba = boa.import('jieba');
+const random = boa.import('random');
+const { MultinomialNB } = boa.import('sklearn.naive_bayes');
+const pickle = boa.import('pickle');
+
+const { open } = boa.builtins();
+
+function strip(str: string): string {
+  return str.replace(/(^\s*)|(\s*$)/g, '');
+}
+
+function isdigit(str: string): boolean {
+  return /^\d+$/.test(str);
+}
+
+export const MakeWordsSet = function(words_file: string): Promise<Set<string>> {
+  const words_set = new Set<string>();
+  const rl = readline.createInterface({
+    input: fs.createReadStream(words_file)
+  });
+  return new Promise((resolve) => {
+    rl.on('line', (line: string) => {
+      const word = strip(line);
+      if (word.length > 0 && !words_set.has(word)) {
+        words_set.add(word);
+      }
+    });
+
+    rl.on('close', () => {
+      resolve(words_set);
+    });
+  });
+};
+
+export const TextProcessing = function(row_data: string[], row_class: string[]): any[][] {
+  const data_list: string[][] = [];
+  const class_list: string[] = [];
+
+  row_data.forEach((data, i) => {
+    const word_cut: string[] = jieba.cut(data, boa.kwargs({
+      cut_all: false
+    }));
+    data_list.push(word_cut);
+    class_list.push(row_class[i]);
+  });
+      
+  // 划分训练集和测试集
+  const data_class_list = zip(data_list, class_list);
+  random.shuffle(data_class_list);
+  const [ train_data_list, train_class_list ] = unzip(data_class_list);
+
+  interface AllWordsDict {
+    [key: string]: [ string, number ];
+  }
```

Comment:
Could you please move the interface on the top level?

Reply:
Of cause~